plugins/woocommerce/tests/e2e-pw/global-setup.js

```diff
@@ -15,6 +16,22 @@ module.exports = async ( config ) => {
 	console.log( `State Dir: ${ stateDir }` );
 	console.log( `Base URL: ${ baseURL }` );
 
+	// Try calling the site reset plugin (if testing on an external environment)
```

Comment:
> Why skipping local? I don't think it would hurt.

Because it just adds time.  `wp-env` is usually freshly started, so there's no reason to start it and then immediately reset it.  

> But TBH I would rather see this part of the envs setup script, and only execute it for the envs we want

The trouble with that approach is that if it fails, that isn't communicated to the test suite/in the test results.  Wrapping this in a `try` will still allow the tests to run even if the reset doesn't succeed, but if it does fail, it will be clear in the output/results that the reset failed and that could be why a particular test failed.

`global-setup` is there to help prepare the test environment, so I'm not sure why we'd avoid using it?

Reply:
> Because it just adds time. wp-env is usually freshly started, so there's no reason to start it and then immediately reset it.

I agree it makes no sense to immediately reset a fresh env.

> The trouble with that approach is that if it fails, that isn't communicated to the test suite/in the test results.

Not necessarily true. As long as the error is not caught and the script returns the proper exit code the setup step will properly fail in CI and the next step will not run because of it. It will be clear why and what failed. Locally, the same, you'll see an error. A failing global-setup is not much different in terms of reporting.

> global-setup is there to help prepare the test environment, so I'm not sure why we'd avoid using it?

I would avoid having this in global-setup for two reasons:
- the conditional part. It doesn't look right to me to have a setup block conditioned by the environment url.
- I cannot skip it. Most probably I don't want a full reset with each local run.

Reply:
In the end, I think neither solution is perfect. I would be fine with having it in global-setup, but can you the option to skip it? Using an environment variable for it will probably be the way here.